packages/ethereum-contracts/test/foundry/superfluid/CFAv1NFTBase.t.sol

```diff
@@ -0,0 +1,434 @@
+// SPDX-License-Identifier: AGPLv3
+pragma solidity 0.8.16;
+
+import { UUPSProxy } from "../../../contracts/upgradability/UUPSProxy.sol";
+import {
+    CFAv1NFTBase,
+    ConstantOutflowNFT
+} from "../../../contracts/superfluid/ConstantOutflowNFT.sol";
+import {
+    ConstantInflowNFT
+} from "../../../contracts/superfluid/ConstantInflowNFT.sol";
+
+import {
+    CFAv1Library,
+    FoundrySuperfluidTester
+} from "../FoundrySuperfluidTester.sol";
+
+contract ConstantOutflowNFTMock is ConstantOutflowNFT {
+    /// @dev a mock mint function that exposes the internal _mint function
+    function mockMint(
+        address _to,
+        address _flowReceiver,
+        uint256 _newTokenId
+    ) public {
+        _mint(_to, _flowReceiver, _newTokenId);
+    }
+
+    /// @dev a mock burn function that exposes the internal _burn function
+    function mockBurn(uint256 _tokenId) public {
+        _burn(_tokenId);
+    }
+
+    /// @dev this ownerOf doesn't revert if _tokenId doesn't exist
+    function mockOwnerOf(uint256 _tokenId) public view returns (address) {
+        return _ownerOf(_tokenId);
+    }
+}
+
+contract ConstantInflowNFTMock is ConstantInflowNFT {
+    /// @dev a mock mint function to emit the mint Transfer event
+    function mockMint(address _to, uint256 _newTokenId) public {
+        _mint(_to, _newTokenId);
+    }
+
+    /// @dev a mock burn function to emit the burn Transfer event
+    function mockBurn(uint256 _tokenId) public {
+        _burn(_tokenId);
+    }
+
+    // @dev this ownerOf doesn't revert if _tokenId doesn't exist
+    function mockOwnerOf(uint256 _tokenId) public view returns (address) {
+        return _ownerOf(_tokenId);
+    }
+
+    /// @dev this exposes the internal flow data by token id for testing purposes
+    function mockFlowDataByTokenId(
+        uint256 _tokenId
+    ) public view returns (FlowData memory flowData) {
+        return flowDataByTokenId(_tokenId);
+    }
+}
+
+abstract contract CFAv1BaseTest is FoundrySuperfluidTester {
```

Comment:
12. Abstract Base Test contract to be inherited

Reply:
13. Assertion Helpers, Helper Functions, Assume Helpers, Passing Cases in this contract